### PR TITLE
Frontend audit: fix React/state/CSS bugs in src/app and src/diagram

### PR DESCRIPTION
**Pull request by bpowers**
## What this is

A systematic bug audit of the frontend TypeScript/React/CSS code in `src/app` and `src/diagram`, followed by fixes for everything confirmed. Eight area/dimension review agents swept the code; every finding was adversarially verified by an independent agent before being accepted (51 raw findings -> 29 confirmed, 10 refuted). The full findings list with per-item status lives in [docs/dev/frontend-audit-2026-06.md](docs/dev/frontend-audit-2026-06.md), committed first on this branch so the fixes can be reviewed against it.

Every fix was done test-first where the behavior was testable (new pure helpers + component tests; 20+ new test files/cases), and each landed as its own commit through the full pre-commit suite.

## Highlights (the bugs behind the glitches you reported)

- **Equation preview glitches**: three compounding causes. (1) The details panels were keyed by `projectVersion`, which bumps on every pan frame and is reset by every autosave completion -- each bump remounted the open panel, discarding in-progress Slate edits and refiring the async LaTeX load. They are now keyed by a new `projectGeneration` that increments only on real content changes (edits, undo/redo). (2) While the engine LaTeX loaded (and whenever the engine couldn't produce LaTeX), the raw equation text was fed to `katex.renderToString`, rendering `revenue_per_unit` with `_` as subscripts; raw text now renders as plain monospace. (3) Error highlighting treated Rust byte offsets as UTF-16 indices and only ever touched line 0; a new `equation-highlight` module converts offsets, shifts past the `{apply-to-all:}` prefix, and highlights across lines (caret placement from preview clicks got the same fix).
- **Long equation running off the card**: KaTeX packs math into atomic inline-block `.base` spans with `white-space: nowrap`, which `overflow-wrap` on an ancestor can't break, and the flex container refused to shrink (`min-width: auto`). Fixed with a `.base` override (higher specificity than katex.css), `min-width: 0`, and `overflow-x: auto` as the unbreakable-content fallback.
- **Search bar / details card misalignment**: `reset.css` (the universal `box-sizing: border-box`) is tree-shaken out of the production CSS bundle even though it's a side-effect import -- so the padded search bar rendered 16px wider than the card. Fixed two ways: an explicit `import '@simlin/diagram/reset.css'` in the app entry (verified present in the rebuilt bundle) and explicit `box-sizing: border-box` on every panel rule so alignment never depends on a global reset. The unexplained tree-shaking is tracked as #708.

## Other high-severity fixes

- **Undo was broken two ways**: editing after an undo kept the abandoned redo branch (undo then jumped to sibling states instead of the edit's parent), and every pan/zoom/momentum frame pushed a snapshot into the 5-entry undo buffer, evicting real edits. View-only updates no longer touch history; real edits go through a pure, tested `advanceProjectHistory`.
- **Creating a flow armed a delete trap**: `flowStillBeingCreated` was never reset (the reset was a `true`-for-`false` typo), so a later Escape-cancel of any rename deleted that variable.
- **Logout did nothing** -- the menu item only closed the menu, and the server's `DELETE /session` was a TODO stub that never even wrote a response. Both halves implemented (Firebase signOut + session invalidation via passport `req.logout()` + seshcookie rewrite, with an end-to-end server test).
- **A failed project load left a permanently blank editor** (network error / non-JSON / missing fields all escaped as unhandled rejections); failures now render an error message.
- **NewUser onboarding could submit via backdrop click or Enter while bypassing the terms-agreement gate**; `Dialog` gained `disableBackdropClick` and the gate is enforced on every entry path.

Plus mediums/lows: Login error handling and stale-error clearing, Autocomplete dropdown detaching on scroll/resize, flow/connector rendering geometry edge cases (fixed identically in the TS *and* Rust renderers -- the SVG parity test compares them byte-for-byte), LookupEditor NaN on a 1-point table, missing React keys, PureComponent memoization defeated by per-render closures, unmount guards, breakpoint-aware panel-width math, Checkbox/TextField contract fixes.

## Decisions and tradeoffs

- **Remount-by-key kept, but on the right key.** Rather than teaching VariableDetails to resync Slate state from props (Slate ignores post-mount `initialValue` changes, so prop-resync wouldn't reach the DOM and would need editor-instance recreation), I kept the constructor-seeding + key-remount design and made the key change exactly when content changes. Lower risk, preserves all existing refresh semantics, and the difference is observable in tests.
- **Undo after a view change restores the camera of the snapshot.** Excluding pans from history means undo now jumps the viewport to where it was at edit time. I judged this standard editor behavior and better than view changes consuming undo slots.
- **Rust renderer changed in a "frontend" PR.** The TS and Rust SVG renderers are parity-tested byte-for-byte, so the geometry fixes had to land in both (the Rust code even carried a "Matches TypeScript: tan, not atan" comment). One hardcoded golden string was updated for atan's rounding on ordinary arcs (third-decimal differences).
- **Unit warnings no longer force the raw editor open.** This matches `variableDetailsView` (unit errors are non-fatal: they keep the chart) -- the warning list still renders under the chart, and clicking into the editor still shows the squiggle.
- **What I deliberately did not do**: no class-to-function component conversion (per your instruction); no debounce of the per-frame engine round-trips (#707 -- removing the undo pollution fixed the worst symptom, persistence batching is a bigger change); dark-mode for the component library (#709) and Menu anchor-tracking (#710) deferred and tracked.

## Verification

- All new and existing tests pass; every commit went through the full pre-commit suite (Rust fmt/clippy/tests, TS lint/build/tsc/tests, WASM build, Python tests).
- The CSS fixes are locked in by stylesheet-contract tests (`panel-css.test.ts`) and the rules were verified present in the rebuilt production bundle. I could not do a pixel-level browser check in this environment (headless Chromium lacks system libs here) -- worth a quick visual confirmation of the reliability model's details panel on your running app.
- 12 of 51 raw findings lost their verifier mid-run; I recovered them from the workflow journal and confirmed they all duplicated already-confirmed findings before proceeding.

Follow-ups filed: #707 (per-frame engine round-trips), #708 (reset.css tree-shaking), #709 (component-library dark mode), #710 (Menu anchor tracking).

### CODE REVIEW

**Comment by bpowers** [comment]
@codex review

File: src/app/App.tsx [hunk 1]
```diff
@@ -245,6 +246,30 @@ export class InnerApp extends React.PureComponent<{}, AppState> {
...
+      });
+    } catch (err) {
+      console.error('logout: clearing the server session failed:', err);
+    }
+    try {
+      await signOut(this.state.auth);
+    } catch (err) {
+      console.error('logout: firebase signOut failed:', err);
+    }
+    await userInfo.invalidate();
```
[the remark this answers is not in the document]
Comment (bpowers) [reply src/app/App.tsx]: Fixed in 344aadc5: local state is now cleared before the cache refresh and the refresh is guarded, matching the handler's best-effort contract. Also fixed one layer deeper -- invalidate() starts a fetch nothing immediately awaits, so UserInfoSingleton's worker now catches network failures and reports status 0 (not authenticated) instead of leaving a rejecting promise to surface later as an unhandled rejection. Regression test added (logout with every network call rejecting still resolves and clears the user).
